src/browser/modules/Editor/styled.jsx

```diff
@@ -24,51 +24,80 @@ import { dim } from 'browser-styles/constants'
 const editorPadding = 10
 
 export const BaseBar = styled.div`
-  background-color: ${props => props.theme.primaryBackground};
-  display: flex;
-  flex-direction: row;
-  align-items: middle;
-  min-height: ${props =>
-    props.expanded
-      ? '100vh'
-      : Math.max(dim.editorbarHeight, props.minHeight + editorPadding * 2)}px;
-  overflow: hidden;
+  background-color: ${props => props.theme.editorBackground};
+  border-radius: 4px;
+  margin: ${editorPadding}px 0px ${editorPadding}px 0;
+  display: grid;
+  // minmax(0, 1fr) prevents the editor from growing the text field
+  grid-template-columns: minmax(0, 1fr) auto;
+  grid-template-areas: ${props => {
+    if (props.expanded || props.card) {
+      return "'header' 'editor'"
+    }
+    return "'editor header'"
+  }};
+`
+
+export const Header = styled.div`
+  grid-area: header;
+  border-radius: 4px 4px 0 0;
+  ${props => {
+    if (props.expanded) {
+      return `background-color: #4d4a57;
+              border-radius: 0;
+      `
+    }
+    if (props.card) {
+      return `background-color: #4d4a57;
+              transition-duration: 0.3s;`
+    }
+    return ''
+  }}
 
-  margin: 0;
+  display: flex;
+  justify-content: flex-end;
 `
+
 export const Bar = styled(BaseBar)`
   ${props => {
     if (props.expanded) {
-      return (
-        'position: absolute;' +
-        'height: 100vh;' +
-        'z-index: 100;' +
-        'right: 0;' +
-        'left: 0;' +
-        'bottom: 0;'
-      )
+      return `
+position: fixed;
+top: -10px;
+bottom: 0;
+left: 0;
+right: 0;
+height: 100vh;
+border-radius: 0;
+z-index: 1030;`
     }
   }};
 `
 export const ActionButtonSection = styled.div`
-  flex: 0 0 120px;
-  align-items: top;
   display: flex;
-  padding-top: 21px;
   justify-content: space-between;
+  width: ${props => props.width}px;
+  margin: 7px;
 `
 
 const BaseEditorWrapper = styled.div`
-  flex: auto;
-  padding: ${editorPadding}px ${editorPadding}px ${editorPadding}px 0;
   font-family: 'Fira Code', Monaco, 'Courier New', Terminal, monospace;
-  min-height: ${props =>
-    props.expanded
-      ? '100vh'
-      : Math.max(dim.editorbarHeight, props.minHeight + editorPadding * 2)}px;
-  width: 0;
+  grid-area: editor;
+
+  min-height: ${props => {
+    if (props.expanded) {
+      return '100vh'
+    }
+    if (props.card) {
+      // 230 is 10 lines + 2*12px padding
+      return '254px'
+    }
+    return '0'
+  }};
+
+  ${props => (props.expanded ? '' : 'transition-duration: 0.1s;')}
+
   .CodeMirror {
-    background-color: ${props => props.theme.editorBackground} !important;
```

Comment:
![Neo4j Browser - Google Chrome 8_4_2020 11_25_35 AM](https://user-images.githubusercontent.com/939458/89277873-aa82ae80-d645-11ea-9691-72b23f5eaa28.png)

Looks off in dark mode

Reply:
Should be fixed now:
<img width="642" alt="image" src="https://user-images.githubusercontent.com/10564538/89308428-88535580-d672-11ea-9cf8-ec25224c1068.png">

I do think the dark mode editor is too black, it could be the same color as the frames? Also, the header color doesn't fit as well in dark mode either

Reply:
Both light and dark mode look correct to me on Windows :)